### DISCUSSION TITLE
Adding Vitessce widgets to a visualization dashboard

### DISCUSSION THREAD

**Discussion by georgezakinih** [vitessce/vitessce-python#346] [General]
Hi Vitessce Python Team,

I'm planning to create a dashboard for interactive, real-time visualization of an AnnData object. The dashboard will consist of multiple tabs, and I'm interested in incorporating Vitessce widgets into one of these tabs.

I'm currently exploring frameworks like Streamlit, Voilà, and Shiny for Python to build this dashboard. I've noticed that the Vitessce ipywidget is compatible with Voilà. Additionally, Streamlit has a large and active community.

I'd appreciate your insights on which framework would be the best fit for integrating with Vitessce.

Thanks,
George

**Comment by keller-mark** [comment]
Hi @georgezakinih ,
I do not have much experience with these dashboard frameworks. However I will note that some of these frameworks are compatible with ipywidgets (and the Vitessce widget in this package is a custom ipywidget) while others are not (I believe streamlit has its own [proprietary widget protocol](https://docs.streamlit.io/develop/concepts/custom-components/intro#create-a-bi-directional-component)) which would require wrapping the Vitessce JavaScript package with that custom protocol. It also depends whether bidirectional communication is needed (if not, you may be able to use the [VitessceConfig.display](https://vitessce.github.io/vitessce-python/api_config.html#vitessce.config.VitessceConfig.display) function which returns ipython HTML output).

**Reply by georgezakinih** [reply]
Thanks for following up Mark and for pointing me to the VitessceConfig.display.

Would you please elaborate what you mean by bidirectional communication?

I have tested that Voilà is compatible with the Vitessce ipywidget. What other frameworks are compatible as well?
